src/lib/dai.c

```diff
@@ -202,6 +205,15 @@ static void dai_set_device_params(struct dai *d)
 		d->dma_dev = DMA_DEV_HDA;
 		d->dma_caps = DMA_CAP_HDA;
 		break;
+	case SOF_DAI_INTEL_UAOL:
+#ifdef CONFIG_DMA_INTEL_ADSP_GPDMA
+		d->dma_dev = DMA_DEV_ALH;
+		d->dma_caps = DMA_CAP_GP_LP | DMA_CAP_GP_HP;
+#else
+		d->dma_dev = DMA_DEV_HDA;
+		d->dma_caps = DMA_CAP_HDA;
+#endif
```

Comment:
same comment, we are not going to add UAOL on older generations, that ship has sailed. So the code with GPDMA is dead already.

Reply:
Removed the code.